Gems/ScriptCanvas/Code/Source/SystemComponent.cpp

```diff
@@ -152,6 +152,15 @@ namespace ScriptCanvas
 
         m_infiniteLoopDetectionMaxIterations = ScriptCanvasSystemComponentCpp::k_infiniteLoopDetectionMaxIterations;
         m_maxHandlerStackDepth = ScriptCanvasSystemComponentCpp::k_maxHandlerStackDepth;
+
+        ScriptCanvas::AutoGenRegistryManager::Init();
+        if (auto componentApplication = AZ::Interface<AZ::ComponentApplicationRequests>::Get())
+        {
+            for (auto descriptor : ScriptCanvas::AutoGenRegistryManager::GetComponentDescriptors())
+            {
+                componentApplication->RegisterComponentDescriptor(descriptor);
+            }
+        }
```

Comment:
Component Descriptors can't be registered in `Component::Init()`

This can occur in [ScriptCanvasModuleCommon::GetRequiredSystemComponents](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/Module.h#L68-L72) for a non-instance location.

Reply:
There is one issue by putting into GetRequiredSystemComponents function, seems like AP doesn't trigger it.
It only works to put in SystemComponent for AP use case

Reply:
Honestly, I think that call to AutoGenRegistryManger::Init() and RegisterComponentDescriptor for the AutoGen nodes should go in the `ScriptCanvasModuleCommon` constructor which would be called ALL applications, that load the ScriptCanvas.

We would just need to split the Module Initialization steps into a `ModuleInitializationSteps::Load`, `ModuleInitializationSteps::CreateClass` and `ModuleInitializationSteps::RegisterComponentDescriptors` split

Reply:
yes, this sounds like a reasonable approach.